documentation/modules/appendix/pages/best-practices.adoc

```diff
@@ -12,56 +12,854 @@ SPDX-License-Identifier: MPL-2.0
-  xref:ROOT:characteristics.adoc#list-characteristic[List],
-  xref:ROOT:characteristics.adoc#set-characteristic[Set] or
-  xref:ROOT:characteristics.adoc#sorted-set-characteristic[Sorted Set] Characteristic (and no other
-  Properties), the Aspect's name should be in plural. Otherwise, if there is exactly one Property in
-  the Aspect that has a xref:ROOT:characteristics.adoc#time-series-characteristic[Time Series]
-  Characteristic, the Aspect's name should end with "History". Otherwise, the Aspect's name should
-  be in singular.
+The following best practices, while not mandatory, are highly recommended for optimal performance and to mitigate potential issues, especially when working with Aspect Models at scale.
+
+[[the-name-attribute]]
+== The `name` attribute
+
+The `name` attribute is mandatory for all model elements.
+
+Use camel-case capitalization, as detailed in the xref:ROOT:modeling-guidelines.adoc#naming-rules[naming rules].
+
+For optimal results, consider the following best practices:
+
+* <<camel-case-for-acronyms>>
+* <<unmistakable-abbreviations>>
+* <<no-redundant-prefix>>
+* <<no-redundant-suffix>>
+* <<meaningful-names>>
+* <<add-payload-names-for-properties>>
+* <<if-certain-characteristics-are-present>>
+
+In addition to that, there are also best practices for:
+
+* <<the-preferredname-attribute>>
+* <<the-description-attribute>>
+
+[[camel-case-for-acronyms]]
+=== Camel case for acronyms
+
+For every part of a `name`, also for acronyms, strictly apply camel-case capitalization as detailed in the  xref:ROOT:modeling-guidelines.adoc#naming-rules[naming rules].
+
+For example:
+
+|===
+|{nok-small} DON'T |{ok-small} DO
+
+|`mainECU`
+|`mainEcu`
+|`plantID`
+|`plantId`
+|`documentURL`
+|`documentUrl`
+|===
+
+Create `name` attributes where each capital letter can be trusted to trigger the start of a new name part.
+
+That way, for example, automated case conversion will produce meaningful names (e.g., when generating code based on an Aspect Model):
+
+|===
+|SAMM `name` |Resulting name in generated code (automated case conversion)
+
+|{nok-small} `documentURL`
+|`DOCUMENT_U_R_L` {nok-small}
+|{ok-small} `documentUrl`
+|`DOCUMENT_URL` {ok-small}
+|===
+
+[[unmistakable-abbreviations]]
+=== Unmistakable abbreviations
+
+Avoid abbreviations that are not sufficiently common.
+Seemingly obvious abbreviations in one context can be misunderstood in different contexts.
+
+For example:
+
+|===
+|{nok-small} DON'T |{ok-small} DO
+
+|`temp`
+|Abbreviation unclear; not recommended. (Could mean "temperature" or "temporary".)
+|Abbreviating “minimum” or “maximum” by `min` or `max` is common enough and, therefore, acceptable.
+|`minTemperature` +
+`maxSpeed`
+|===
+
+Using unmistakable abbreviations also applies to <<abbreviations-in-descriptions,Description texts>>.
+
+[[no-redundant-prefix]]
+=== No redundant prefix
+
+Do not duplicate information.
+
+Avoid prefixes that repeat a domain (the namespace) an Aspect Model belongs to.
```

Comment:
only for aspect model? same for Entity, no?

However, there are cases in which this is necessary. For example if you have the property "name" in two different Entities. SAMM requires that all names in a namespace are unique.

Then you should add the prefix of the Entity but define a corresponding payload name without the prefix.